reducing_utils.go

```diff
@@ -74,23 +91,26 @@ References:
 	http://www.oasis-open.org/committees/documents.php?wg_abbrev=office-formula
 	OpenDocument-formula-20090508.odt
 */
-func IPmt(rate float64, per int64, nper int64, pv float64, fv float64, when paymentperiod.Type) float64 {
+func IPmt(rate decimal.Decimal, per int64, nper int64, pv int64, fv int64, when paymentperiod.Type) *decimal.Decimal {
+	// TODO: update nan and remove any rounding here.
 	totalPmt := Pmt(rate, nper, pv, fv, when)
-	ipmt := rbl(rate, per, totalPmt, pv, when) * rate
+	one := decimal.NewFromInt(1)
+	ipmt := rbl(rate, per, totalPmt.IntPart(), pv, when).Mul(rate)
```

Comment:
```suggestion
	ipmt := rbl(rate, per, totalPmt.Round(0).IntPart(), pv, when).Mul(rate)
```
@krantideep95 here, instead of passing only the int part, this can be rounded to nearest int. Other option is to use replace int64 with decimal everywhere and pass it as it is. thoughts ?

Reply:
replace int64 with decimal everywhere.